crates/transaction-pool/src/metrics.rs

```diff
@@ -13,4 +13,23 @@ pub struct TxPoolMetrics {
     pub(crate) invalid_transactions: Counter,
     /// Number of removed transactions from the pool
     pub(crate) removed_transactions: Counter,
+    /// Total number of transactions in the pool
+    pub(crate) total_number_transactions: Gauge,
```

Comment:
I think this can be inferred using `inserted_transactions - removed_transactions - invalid_transactions`, so no need for a separate metric.

Alternatively, it can be inferred using `pending_pool_length + basefee_pool_length + queued_pool_length` etc.

Reply:
Thanks for the review @onbjerg, it's very instructive! :)
I renamed the metrics, removed the ones that can be inferred and updated the documentation.